packages/features/ee/managed-event-types/hooks/useLockedFieldsManager.tsx

```diff
@@ -32,28 +71,68 @@ const useLockedFieldsManager = (
     eventType.metadata?.managedEventConfig !== undefined &&
     eventType.schedulingType !== SchedulingType.MANAGED;
 
-  const shouldLockIndicator = (fieldName: string) => {
+  const setUnlockedFields = (fieldName: string, val: boolean | undefined) => {
+    const path = "metadata.managedEventConfig.unlockedFields";
+    const metaUnlockedFields = formMethods.getValues(path);
+    if (!metaUnlockedFields) return;
+    if (val === undefined) {
+      delete metaUnlockedFields[fieldName as keyof typeof metaUnlockedFields];
+      formMethods.setValue(path, { ...metaUnlockedFields });
+    } else {
+      formMethods.setValue(path, {
+        ...metaUnlockedFields,
+        [fieldName]: val,
+      });
+    }
+  };
```

Comment:
In charge of using form methods to store the unlocked fields in the event type metadata. When locked, it gets removed.